vms/platformvm/state/state_test.go

```diff
@@ -1401,6 +1401,28 @@ func TestStateSubnetOwner(t *testing.T) {
 	require.Equal(owner2, owner)
 }
 
+func TestStateSubnetManager(t *testing.T) {
```

Comment:
1. One thing we could test here is the cache usage.
2. Another interesting edge case to consider is: "what is the maximum Address size?"

Reply:
Added a cache test.

For the maximum address size, I think bounding it at `4096` is more than enough. [ML-DSA-87](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.204.ipd.pdf) (previously known as DILITHIUM) has a public key size of 2592. I don't think this should be bounded on the `state` interface and should instead be bounded on the `ConvertSubnetTx` transaction.